walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,62 +14,77 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
-    
+  
+  /** First Solution: 
+   * Separate {@code events} in two lists based on attendee type.
+   * Call .queryHelper() on each list
+   * Return a collection with valid time slots accroding to requesst.
+   *
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+
     // If the request has no attendees, the open time slot is the entire day.
     if (request.getAttendees().isEmpty() && 
         request.getOptionalAttendees().isEmpty()) {
         return Arrays.asList(TimeRange.WHOLE_DAY);
     }
 
     // Sort {@code events} by starting time.
-    List<Event> eventsList = new ArrayList(events);
+    List<Event> eventsList = new ArrayList<>(events);;
+
     eventsList.sort((Event a, Event b) -> {
         return TimeRange.ORDER_BY_START.compare(a.getWhen(), b.getWhen());
     });
 
     // Filter out events with attendees not in {@code request}.
-    if (request.getOptionalAttendees() != null){ 
     eventsList = eventsList.stream()
-                           .filter(e -> request.getAttendees()
-                                               .containsAll(e.getAttendees()) || 
-                                        request.getOptionalAttendees()
-                                               .containsAll(e.getAttendees()))
-                           .collect(Collectors.toList());
-    } else {
-        eventsList = eventsList.stream()
-                           .filter(e -> request.getAttendees()
-                                               .containsAll(e.getAttendees()))
+                           .filter(event -> !Collections.disjoint(request.getAttendees(), event.getAttendees()) || 
+                                        !Collections.disjoint(request.getOptionalAttendees(), event.getAttendees()))
                            .collect(Collectors.toList());
-    }
     List<Event> mandatoryEvents = eventsList.stream()
-                           .filter(e -> request.getAttendees()
-                                               .containsAll(e.getAttendees()))
+                           .filter(event -> !Collections.disjoint(request.getAttendees(), event.getAttendees()))
+                           .collect(Collectors.toList());
+    Collection<Event> optionalEvents = eventsList.stream()
```

Comment:
Should we make this List<Event>?

Reply:
Sure, the reason I left it a Collection was just so everything was the same.

Reply:
Sure I had it as a Collection at first just to avoid any type-conversion errors.